graphbolt/src/csc_sampling_graph.cc

```diff
@@ -123,11 +123,46 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
 
 c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
     const torch::Tensor& nodes) const {
-  // TODO(#5692): implement this.
+  const int64_t num_nodes = nodes.size(0);
+
+  std::vector<torch::Tensor> picked_neighbors_per_node(num_nodes);
```

Comment:
Do you know if there is a performance gap between `std::vector` and `torch::Tensor`? DGL uses a solution without `vector`, which creates a tensor for the sampled `indptr` directly.